### PR TITLE
DDPB-4009 keep images updated

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
Update all our images so they pass the ECR scans and make ECR scan notify us if they get sec vulnerabilities again (as there was a syntax error preventing notifications)

Fixes DDPB-4009

## Approach
So there's a few interesting things going on here in this tiny PR which caused me some headache. It all boils down to having latest version of everything in the end though there were points I went off on completely different paths and went into the weeds with python... Also went into the weeds with alpine packages.. in the end it was more straight forward.

First thing! Locally you need at least docker desktop 3.3 on mac 
Next circle remote docker needs to be 20.10

This is so we can even build alpine image 3.14 (as security has been tightened up) which both the new php and python images i've used use as the base image.

Next i changed the slim buster images to alpine and used an alpine add package for wkhtmltopdf that adds the same 12.06 version but for alpine yay. It looked like that was adding the font stuff but it wasn't so needed to use some other apk font packages.

cron is a bit different in alpine.

Everything is a bit pinned and this seems like a good idea with base packages as we've had them break before and they're hard to work out sometimes whats gone wrong.

Lastly... weird issue with pdf gen where it says LEAK in the logs.. However it works fine. Read up that this is just a warning message but I want to keep an eye and try a few runs of behat on it before we merge... 

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: wkhtmltopdf/docker-entrypoint.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
-service cron restart
+crond -l 2
```
Comment (acsauk) [inline wkhtmltopdf/docker-entrypoint.sh:2]: Is this essentially doing the same thing as before or does 2 correspond to a corn period?

**Review by acsauk — COMMENTED** [review]
Couple of Qs but looks good to go - I'm assuming now we're pinning we'll rely on the ECR scan results on when we need to bump the version?

File: wkhtmltopdf/docker-entrypoint.sh (hunk 1 above)
Comment (jamesrwarren) [reply wkhtmltopdf/docker-entrypoint.sh:2]: Yeah this is basically the log level for crond which only alerts on critical alerts. crond is the alpine version of the cron daemon. service isn't a thing in alpine. Alpine containers ahve their own cron daemon that can be activated by default and the standard folders it picks up from are different too, hence the change to those folders :)

I tested this locally by adding a script in for it to run every 15 mins and checked it was working. It runs the crond as a sub process of the main .sh process

File: wkhtmltopdf/Dockerfile [hunk 2]
```diff
@@ -1,23 +1,19 @@
-FROM python:3.9-rc-slim-buster
+FROM python:3.10-rc-alpine3.14
```
[the remark this answers is not in the document]
Comment (jamesrwarren) [reply wkhtmltopdf/Dockerfile:1]: This is pretty pinned down. This means using 3.10 of python and 3.14 of alpine image. the 3rd part of semantic versioning (patch) really shouldn't break anything and so i think it's fine to have it upgrade but the 'minor' part defo can break things as we have seen repeatedly before so i was thinking it needs to be updated manually when we get ECR alerts.

For example the difference between alpine 3.13 and 3.14 is big enough for it not to build at all on older version of docker client.

[the remark this answers is not in the document]
Comment (acsauk) [reply wkhtmltopdf/Dockerfile:1]: Ah cool OK that makes sense 👍

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**